### PR TITLE
Anko - Change Search screen from xml to anko layout

### PR DESCRIPTION
**Pull request by at-cuongcao**
#### Related issues (Required)
- Task #112 

#### Description (Required)
- [x] Change xml to anko layout

### CODE REVIEW

File: app/src/main/java/vn/asiantech/way/ui/search/SearchActivity.kt [hunk 1]
```diff
@@ -0,0 +1,72 @@
+package vn.asiantech.way.ui.search
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/SearchActivity.kt]: WARNING: Multiple occurrences of the same string literal within a single kt file detected.

Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/SearchActivity.kt]: WARNING: Multiple occurrences of the same string literal within a single kt file detected.

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+    var onItemClick: (location: Location) -> Unit = {}
+
+    override fun getItemCount() = locations.size
+
+    override fun onBindViewHolder(holder: LocationViewHolder?, position: Int) {
+        holder?.onBind()
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): LocationViewHolder? {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        return LocationAdapterUI().createView(AnkoContext.Companion.create(parent.context,
+                parent, false)).tag as? LocationViewHolder
+    }
+
+    /**
+     * View holder of RecyclerView's item.
+     */
+    inner class LocationViewHolder(itemView: View,
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: WARNING: Unexpected indentation

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+                                   private val imgLocationIcon: ImageView,
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: WARNING: Unexpected indentation

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+                                   private val tvLocationName: TextView,
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: WARNING: Unexpected indentation

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+    var onItemClick: (location: Location) -> Unit = {}
+
+    override fun getItemCount() = locations.size
+
+    override fun onBindViewHolder(holder: LocationViewHolder?, position: Int) {
+        holder?.onBind()
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): LocationViewHolder? {
+        return LocationAdapterUI().createView(AnkoContext.Companion.create(parent.context,
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+                            padding = dimen(R.dimen.small_padding_margin)
+                            textSizeDimen = R.dimen.search_screen_text_size
+                        }.lparams(matchParent, wrapContent) {
+                            margin = dimen(R.dimen.default_padding_margin)
+                            rightOf(ID_IMG_CHOOSE_ON_MAP_ICON)
+                        }
+                    }.lparams(matchParent, wrapContent)
+
+                    recyclerView {
+                        id = ID_RECYCLER_VIEW_LOCATIONS
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt]: Change id to `ID_RECYCLER_VIEW_LOCATION`

File: app/src/main/java/vn/asiantech/way/ui/search/SearchActivity.kt [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+class SearchActivity : BaseActivity() {
+
+    private lateinit var searchActivityUI: SearchActivityUI
+    private lateinit var adapter: LocationAdapter
+    private var locations = mutableListOf<Location>()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        adapter = LocationAdapter(locations)
+        adapter.onItemClick = {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/search/SearchActivity.kt]: move this `click handle` to `SearchActivityUI`

**Comment by at-vinhhuynh** [comment]
@at-tienhoang 
I finish review PR, please check it again.
thank you

File: app/src/main/java/vn/asiantech/way/ui/search/SearchActivity.kt [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+        ui = SearchActivityUI(locations)
+        ui.setContentView(this)
+    }
+
+    override fun onBindViewModel() {
+        // TODO: 30/11/2017
+        // Init later.
+    }
+
+
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/SearchActivity.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+
+    var onItemClick: (location: WayLocation) -> Unit = {}
+
+    override fun getItemCount() = locations.size
+
+    override fun onBindViewHolder(holder: LocationViewHolder?, position: Int) {
+        holder?.onBind()
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): LocationViewHolder? {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+        return LocationAdapterUI().createView(AnkoContext.Companion.create(parent.context,
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/ui/search/SearchActivity.kt (hunk 9 above)
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/SearchActivity.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt (hunk 10 above)
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt (hunk 11 above)
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+            }
+        }
+    }
+
+    /**
+     * Item layout of RecyclerView.
+     */
+    inner class LocationAdapterUI : AnkoComponent<ViewGroup> {
+
+        private lateinit var imgLocationIcon: ImageView
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/search/LocationAdapter.kt]: please remove 3 variables

File: app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt [hunk 13]
```diff
@@ -0,0 +1,146 @@
...
+                            padding = dimen(R.dimen.small_padding_margin)
+                            textSizeDimen = R.dimen.search_screen_text_size
+                        }.lparams(matchParent, wrapContent) {
+                            margin = dimen(R.dimen.default_padding_margin)
+                            rightOf(R.id.search_activity_ui_tv_choose_on_map_icon)
+                        }
+                    }.lparams(matchParent, wrapContent)
+
+                    recyclerView {
+                        id = R.id.search_activity_ui_tv_recycler_view_location
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/search/SearchActivityUI.kt]: change id to `search_activity_ui_recycler_view_location`

File: app/src/main/res/values/ids.xml [hunk 14]
```diff
@@ -28,4 +28,15 @@
     <!-- View ids for CountryAdapter -->
     <item name="country_adapter_tv_tel" type="id" />
     <item name="country_adapter_img_flag" type="id" />
+    <!-- id view on class LocationAdapterUI -->
```
Comment (at-vinhhuynh) [inline app/src/main/res/values/ids.xml]: Change comment to `View ids for LocationAdapterUI ` and below also to `View ids for SearchActivityUI `

**Comment by at-vinhhuynh** [comment]
@at-tienhoang 
I finish review PR, please check it again.
thank you!

**Pull request merged**